### PR TITLE
Translations: Morph `placeholder_translations.py` into a more general script to post-process the "source ts" files generated by `lupdate`

### PR DESCRIPTION
**Pull request by cbjeukendrup**
Resolves: the misery mentioned in https://github.com/musescore/MuseScore/pull/20641#issuecomment-1868644804

### CODE REVIEW

File: tools/translations/process_source_ts_files.py [hunk 1]
```diff
@@ -137,42 +164,48 @@ def tr_error(message_element, description, resolution, hint=''):
...
+            tr_txt = re.sub(r'%([1-9]+)', r'⌜%\1⌝', tr_txt)
+
+            # identify start and end of translatable string
+            tr_txt = '«' + tr_txt + '»'
+
+            if plurals:
+                for idx, plural in enumerate(plurals):
+                    plural.text = 'ᵗʳ' + superscript_numbers[idx] + tr_txt
+            else:
+                translation.text = 'ᵗʳ' + tr_txt
```
Comment (shoogle) [inline tools/translations/process_source_ts_files.py:208]: I think you're missing:
```Python
write_ts_file(placeholder_translations_file, root)
```

**Review by shoogle — APPROVED** [review]
Looks good except you forgot to write to the placeholder file.

Watch out what happens when you upload these files to Transifex. Hopefully the "translations" will be ignored, including in English US and GB on Transifex.

File: tools/translations/process_source_ts_files.py (hunk 1 above)
Comment (cbjeukendrup) [reply tools/translations/process_source_ts_files.py:202]: Oops yes :)

**Comment by cbjeukendrup** [comment]
> Watch out what happens when you upload these files to Transifex. Hopefully the "translations" will be ignored, including in English US and GB on Transifex.

Yes, I'm going to be careful. Will do a tx_pull.sh first, in order to have a backup.

**Comment by shoogle** [comment]
Because you [ran lupdate with `-no-obsolete`](https://github.com/musescore/MuseScore/pull/20710/commits/66df537676fc69ab70e76b5074fab170d8b63139), that means the "synchroni<b><i>s</i></b>e" strings have been replaced with "synchroni<b><i>z</i></b>e" instead of keeping both. Is the plan to run the script again on the 4.2.1 branch after this is merged, this time without `-no-obsolete`?

---

BTW, when `-no-obsolete` is __not__ given, the old "synchroni<b><i>s</i></b>e" strings are retained, but lupdate gives the translation a `type="obsolete"` flag, and of course they no longer have a `<location filename="..." line="..."/>` element since they no longer appear in the source code.

It's not clear what Transifex will do with obsolete strings. If it doesn't like them, you may need to add:

```Python
if translation.get('type') == 'obsolete':
    translation.set('type', 'unfinished')
```
Maybe even:

```Python
if message.find('location') is None:
    location = ET.Element('location', attrib={'filename': 'obsolete.ui', 'line': '1'})
    location.tail = source.tail
    message.insert(0, location)
```

After translation, if the TS files we get back from Transifex contain strings with `type="obsolete"` then those strings will be ignored by `lrelease` and won't appear in the final QM files. (You can check this by [back-converting from QM to TS](https://stackoverflow.com/a/33310460).)

So even if Transifex can cope with obsolete strings, `lrelease` cannot, so it's probably best to add these snippets anyway to ensure that we don't get any `type="obsolete"` flags in the strings we fetch with `tx_pull`.

**Review by igorkorsukov — APPROVED** [review]

**Comment by cbjeukendrup** [comment]
@shoogle It was indeed intentional to run with `-no-obsolete` on the master branch, and without it on 4.2.1. I've applied your suggestion about marking obsolete strings as unfinished, and that seems to make `lrelease` respect them. Next step is to see what happens when I push things to Transifex.

**Comment by cbjeukendrup** [comment]
I will merge this to master, since it has basically been approved and I have verified that the 4.2.1 version (for which I will create a PR soon) works well with Transifex and `lrelease`.

**Pull request merged**